torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -203,8 +203,13 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
   }
 
   // Subclasses must override this method to return the backend name
-  virtual const std::string getBackendName() const = 0;
+  virtual const std::string getBackendName() const {
+    TORCH_INTERNAL_ASSERT(false, "getBackendName is not implemented.");
+  };
```

Comment:
Why do we need this change?

Reply:
Just when registering a class/type with TorchScript, it doesn't like pure virtual method in the class.

Reply:
cc @suo Is there a way to fix this? It doesn't seem right that we have to change a core interface method like this from pure virtual for TorchScript.